tool/test-annocheck.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh -eu
+# Run the `tool/test-annocheck.sh [binary files]` to check security issues.
+# E.g. `tool/test-annocheck.sh ruby libruby.so.3.2.0`.
+#
+# Use annocheck.
+# https://sourceware.org/annobin/
+
+set -x
+
+DOCKER="$(command -v docker || command -v podman)"
+TAG=ruby-fedora-annocheck
+TOOL_DIR=$(dirname "${0}")
+
+"${DOCKER}" build --rm -t "${TAG}" ${TOOL_DIR}/annocheck/
+"${DOCKER}" run --rm -t -v $(pwd):/work "${TAG}" bash -cx "cd /work; annocheck --verbose ${*}"
```

Comment:
Maybe `-w` option works?

```suggestion
"${DOCKER}" run --rm -t -v $(pwd):/work "${TAG}" -w /work annocheck --verbose "$@"
```

Reply:
Thanks for your review! Good point! the `WORKDIR /work` is already in `tool/annocheck/Dockerfile`. So, maybe `docker -w` option is not necessary. Perhaps, we may be able to remove `cd /work;`. Sorry for that. I will check it.

And you are right about "$@". I was able to understand the difference.

https://unix.stackexchange.com/questions/129072/whats-the-difference-between-and
> $* is a single string, whereas $@ is an actual array.

I will fix it.

Reply:
This seems to run the built binary on the Fedora image.
Is it binary compatible with Ubuntu which is used in Github Actions?

Reply:
This line is to use a binary `./ruby` (= `$@`) built on the host os: Ubuntu focal , then the files in the host os's are mounted on the guest os: fedora by `-v $(pwd):/work` (= `-v HOST-DIR:CONTAINER-DIR)` . The `annobin`, the security tool is built on Fedora. But I don't think it is a problem.

Reply:
So, the binary `./ruby` built on Ubuntu is binary compatible with Ubuntu.
If you asking if the binary `annobin` compatibility built on Fedora is compatible with Ubuntu.., here is a document I know about this topic. But I cannot understand the content... There are some limits.

https://www.redhat.com/en/blog/limits-compatibility-and-supportability-containers
> It’s easy to assume that there is nothing to worry about when mixing and matching the container image userspace and host operating system. This post intends to give a realistic explanation on the limits of compatibility with container images
> ...
> So, what are the limits of compatibility?
>
> That’s a tricky question. Let’s level set. Most applications that you want to use in a container will just work.
> ...
> But, there are some limits... 

Actually there are some incompatibilities between different versions of RHEL too. That means there may be incompatibilities between different versions of Ubuntu too.
https://access.redhat.com/support/policy/rhel-container-compatibility

Reply:
I was uncertain if the shared libraries used by built `./ruby` are compatible.
It's ok if it is working.

Reply:
To check if it is working, I want to confirm the new `gcc-11 annocheck` case passes with this patch: https://github.com/ruby/ruby/pull/5895 , and with minimal gcc flags to pass `annocheck` before merging this PR.

Reply:
The gcc-11 annocheck passes now.